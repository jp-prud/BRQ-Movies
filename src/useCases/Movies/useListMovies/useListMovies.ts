import { MoviesService } from '@services';
import { useQuery } from '@tanstack/react-query';
import { QueryKeys } from '@types';

export function useListMovies() {
  const { listMovies } = MoviesService();

  const { data, isLoading, isError, refetch } = useQuery({
    queryKey: [QueryKeys.listMovies],
    queryFn: () => listMovies(),
  });

  return {
    refetch,
    moviesData: {
      list: data,
      isLoading,
      isError,
    },
  };
}
