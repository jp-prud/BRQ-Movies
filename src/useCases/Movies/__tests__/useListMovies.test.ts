import { renderHook, waitFor } from '@tests';

import { useListMovies } from '../useListMovies/useListMovies';

describe('useListMovies', () => {
  it('should list movies', async () => {
    const { result } = renderHook(() => useListMovies());

    await waitFor(() =>
      expect(result.current.moviesData.isLoading).toBe(false),
    );

    expect(result.current.moviesData.isLoading).toBe(false);
    expect(result.current.moviesData).toMatchSnapshot();
  });
});
