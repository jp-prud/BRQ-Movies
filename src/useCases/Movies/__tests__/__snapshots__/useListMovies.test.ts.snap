// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useListMovies should list movies 1`] = `
{
  "isError": false,
  "isLoading": false,
  "list": {
    "data": [
      {
        "adult": false,
        "backdropPath": "https://image.tmdb.org/t/p/w185/tncbMvfV0V07UZozXdBEq4Wu9HH.jpg",
        "genreIds": [
          28,
          80,
          53,
          35,
        ],
        "genres": undefined,
        "id": 573435,
        "originalLanguage": "en",
        "originalTitle": "Bad Boys: Ride or Die",
        "overview": "After their late former Captain is framed, Lowrey and Burnett try to clear his name, only to end up on the run themselves.",
        "popularity": 7285.79,
        "posterPath": "https://image.tmdb.org/t/p/w185/nP6RliHjxsz4irTKsxe8FRhKZYl.jpg",
        "releaseDate": "2024-06-05",
        "title": "Bad Boys: Ride or Die",
        "video": false,
        "voteAverage": 7.653,
        "voteCount": 1132,
      },
      {
        "adult": false,
        "backdropPath": "https://image.tmdb.org/t/p/w185/9l1eZiJHmhr5jIlthMdJN5WYoff.jpg",
        "genreIds": [
          28,
          35,
          878,
        ],
        "genres": undefined,
        "id": 533535,
        "originalLanguage": "en",
        "originalTitle": "Deadpool & Wolverine",
        "overview": "A listless Wade Wilson toils away in civilian life with his days as the morally flexible mercenary, Deadpool, behind him. But when his homeworld faces an existential threat, Wade must reluctantly suit-up again with an even more reluctant Wolverine.",
        "popularity": 7220.83,
        "posterPath": "https://image.tmdb.org/t/p/w185/8cdWjvZQUExUUTzyp4t6EDMubfO.jpg",
        "releaseDate": "2024-07-24",
        "title": "Deadpool & Wolverine",
        "video": false,
        "voteAverage": 8.066,
        "voteCount": 801,
      },
      {
        "adult": false,
        "backdropPath": "https://image.tmdb.org/t/p/w185/stKGOm8UyhuLPR9sZLjs5AkmncA.jpg",
        "genreIds": [
          16,
          10751,
          12,
          35,
        ],
        "genres": undefined,
        "id": 1022789,
        "originalLanguage": "en",
        "originalTitle": "Inside Out 2",
        "overview": "Teenager Riley's mind headquarters is undergoing a sudden demolition to make room for something entirely unexpected: new Emotions! Joy, Sadness, Anger, Fear and Disgust, who’ve long been running a successful operation by all accounts, aren’t sure how to feel when Anxiety shows up. And it looks like she’s not alone.",
        "popularity": 4207.244,
        "posterPath": "https://image.tmdb.org/t/p/w185/vpnVM9B6NMmQpWeZvzLvDESb2QY.jpg",
        "releaseDate": "2024-06-11",
        "title": "Inside Out 2",
        "video": false,
        "voteAverage": 7.635,
        "voteCount": 2143,
      },
      {
        "adult": false,
        "backdropPath": "https://image.tmdb.org/t/p/w185/5g2n3ilC8DpYv4diJeuQ1vKG2Kb.jpg",
        "genreIds": [
          16,
          10751,
          35,
          28,
        ],
        "genres": undefined,
        "id": 519182,
        "originalLanguage": "en",
        "originalTitle": "Despicable Me 4",
        "overview": "Gru and Lucy and their girls — Margo, Edith and Agnes — welcome a new member to the Gru family, Gru Jr., who is intent on tormenting his dad. Meanwhile, Gru faces a new nemesis in Maxime Le Mal and his femme fatale girlfriend Valentina, forcing the family to go on the run.",
        "popularity": 3400.94,
        "posterPath": "https://image.tmdb.org/t/p/w185/wWba3TaojhK7NdycRhoQpsG0FaH.jpg",
        "releaseDate": "2024-06-20",
        "title": "Despicable Me 4",
        "video": false,
        "voteAverage": 7.198,
        "voteCount": 456,
      },
      {
        "adult": false,
        "backdropPath": "https://image.tmdb.org/t/p/w185/dsGwCEO8tda4FlgHKvL95f0oQbH.jpg",
        "genreIds": [
          16,
          878,
          28,
        ],
        "genres": undefined,
        "id": 1209290,
        "originalLanguage": "en",
        "originalTitle": "Justice League: Crisis on Infinite Earths Part Three",
        "overview": "Now fully revealed as the ultimate threat to existence, the Anti-Monitor wages an unrelenting attack on the surviving Earths that struggle for survival in a pocket universe. One by one, these worlds and all their inhabitants are vaporized! On the planets that remain, even time itself is shattered, and heroes from the past join the Justice League and their rag-tag allies against the epitome of evil. But as they make their last stand, will the sacrifice of the superheroes be enough to save us all?",
        "popularity": 1656.771,
        "posterPath": "https://image.tmdb.org/t/p/w185/a3q8NkM8uTh9E23VsbUOdDSbBeN.jpg",
        "releaseDate": "2024-07-16",
        "title": "Justice League: Crisis on Infinite Earths Part Three",
        "video": false,
        "voteAverage": 7.718,
        "voteCount": 103,
      },
      {
        "adult": false,
        "backdropPath": "https://image.tmdb.org/t/p/w185/Akv9GlCCMrzcDkVz4ad8MdLl9DK.jpg",
        "genreIds": [
          16,
          35,
          10751,
          12,
        ],
        "genres": undefined,
        "id": 748783,
        "originalLanguage": "en",
        "originalTitle": "The Garfield Movie",
        "overview": "Garfield, the world-famous, Monday-hating, lasagna-loving indoor cat, is about to have a wild outdoor adventure! After an unexpected reunion with his long-lost father – scruffy street cat Vic – Garfield and his canine friend Odie are forced from their perfectly pampered life into joining Vic in a hilarious, high-stakes heist.",
        "popularity": 1614.576,
        "posterPath": "https://image.tmdb.org/t/p/w185/p6AbOJvMQhBmffd0PIv0u8ghWeY.jpg",
        "releaseDate": "2024-04-30",
        "title": "The Garfield Movie",
        "video": false,
        "voteAverage": 7.205,
        "voteCount": 620,
      },
      {
        "adult": false,
        "backdropPath": "https://image.tmdb.org/t/p/w185/en971MEXui9diirXlogOrPKmsEn.jpg",
        "genreIds": [
          28,
          12,
          35,
        ],
        "genres": undefined,
        "id": 293660,
        "originalLanguage": "en",
        "originalTitle": "Deadpool",
        "overview": "The origin story of former Special Forces operative turned mercenary Wade Wilson, who, after being subjected to a rogue experiment that leaves him with accelerated healing powers, adopts the alter ego Deadpool. Armed with his new abilities and a dark, twisted sense of humor, Deadpool hunts down the man who nearly destroyed his life.",
        "popularity": 1576.742,
        "posterPath": "https://image.tmdb.org/t/p/w185/3E53WEZJqP6aM84D8CckXx4pIHw.jpg",
        "releaseDate": "2016-02-09",
        "title": "Deadpool",
        "video": false,
        "voteAverage": 7.619,
        "voteCount": 30303,
      },
      {
        "adult": false,
        "backdropPath": "https://image.tmdb.org/t/p/w185/iIvjwrDPQHCU4NjbbKpNs88uk6G.jpg",
        "genreIds": [
          28,
          35,
        ],
        "genres": undefined,
        "id": 1048241,
        "originalLanguage": "en",
        "originalTitle": "My Spy The Eternal City",
        "overview": "JJ, a veteran CIA agent, reunites with his protégé Sophie, in order to prevent a catastrophic nuclear plot targeting the Vatican.",
        "popularity": 1533.287,
        "posterPath": "https://image.tmdb.org/t/p/w185/Bf3vCfM94bSJ1saZlyi0UW0e0U.jpg",
        "releaseDate": "2024-07-18",
        "title": "My Spy The Eternal City",
        "video": false,
        "voteAverage": 6.8,
        "voteCount": 157,
      },
      {
        "adult": false,
        "backdropPath": "https://image.tmdb.org/t/p/w185/fqv8v6AycXKsivp1T5yKtLbGXce.jpg",
        "genreIds": [
          878,
          12,
          28,
        ],
        "genres": undefined,
        "id": 653346,
        "originalLanguage": "en",
        "originalTitle": "Kingdom of the Planet of the Apes",
        "overview": "Several generations following Caesar's reign, apes – now the dominant species – live harmoniously while humans have been reduced to living in the shadows. As a new tyrannical ape leader builds his empire, one young ape undertakes a harrowing journey that will cause him to question all he's known about the past and to make choices that will define a future for apes and humans alike.",
        "popularity": 1362.39,
        "posterPath": "https://image.tmdb.org/t/p/w185/gKkl37BQuKTanygYQG1pyYgLVgf.jpg",
        "releaseDate": "2024-05-08",
        "title": "Kingdom of the Planet of the Apes",
        "video": false,
        "voteAverage": 7.172,
        "voteCount": 2152,
      },
      {
        "adult": false,
        "backdropPath": "https://image.tmdb.org/t/p/w185/wNAhuOZ3Zf84jCIlrcI6JhgmY5q.jpg",
        "genreIds": [
          28,
          12,
          878,
        ],
        "genres": undefined,
        "id": 786892,
        "originalLanguage": "en",
        "originalTitle": "Furiosa: A Mad Max Saga",
        "overview": "As the world fell, young Furiosa is snatched from the Green Place of Many Mothers and falls into the hands of a great Biker Horde led by the Warlord Dementus. Sweeping through the Wasteland they come across the Citadel presided over by The Immortan Joe. While the two Tyrants war for dominance, Furiosa must survive many trials as she puts together the means to find her way home.",
        "popularity": 1337.854,
        "posterPath": "https://image.tmdb.org/t/p/w185/iADOJ8Zymht2JPMoy3R7xceZprc.jpg",
        "releaseDate": "2024-05-22",
        "title": "Furiosa: A Mad Max Saga",
        "video": false,
        "voteAverage": 7.655,
        "voteCount": 2380,
      },
      {
        "adult": false,
        "backdropPath": "https://image.tmdb.org/t/p/w185/dn3gbDpXPSwC6saMJOHkCiFA9jn.jpg",
        "genreIds": [
          14,
          12,
          10751,
          35,
        ],
        "genres": undefined,
        "id": 974262,
        "originalLanguage": "en",
        "originalTitle": "Descendants: The Rise of Red",
        "overview": "After the Queen of Hearts incites a coup on Auradon, her rebellious daughter Red and Cinderella's perfectionist daughter Chloe join forces and travel back in time to try to undo the traumatic event that set Red's mother down her villainous path.",
        "popularity": 1125.012,
        "posterPath": "https://image.tmdb.org/t/p/w185/8fYluTtB3b3HKO7KQa5tzrvGaps.jpg",
        "releaseDate": "2024-07-11",
        "title": "Descendants: The Rise of Red",
        "video": false,
        "voteAverage": 7.102,
        "voteCount": 205,
      },
      {
        "adult": false,
        "backdropPath": "https://image.tmdb.org/t/p/w185/3P52oz9HPQWxcwHOwxtyrVV1LKi.jpg",
        "genreIds": [
          28,
          35,
          12,
        ],
        "genres": undefined,
        "id": 383498,
        "originalLanguage": "en",
        "originalTitle": "Deadpool 2",
        "overview": "Wisecracking mercenary Deadpool battles the evil and powerful Cable and other bad guys to save a boy's life.",
        "popularity": 1069.328,
        "posterPath": "https://image.tmdb.org/t/p/w185/to0spRl1CMDvyUbOnbb4fTk3VAd.jpg",
        "releaseDate": "2018-05-10",
        "title": "Deadpool 2",
        "video": false,
        "voteAverage": 7.491,
        "voteCount": 17363,
      },
      {
        "adult": false,
        "backdropPath": "https://image.tmdb.org/t/p/w185/9Y9wTgrMk4G1YNPDDekNk7dybwx.jpg",
        "genreIds": [
          28,
          12,
          10752,
        ],
        "genres": undefined,
        "id": 1152624,
        "originalLanguage": "en",
        "originalTitle": "3 Days in Malay",
        "overview": "Marines stationed at an airfield in Malay during WWII get wind of a coming raid by the Japanese. Unable to get reinforcements approved, they engage in a harrowing 3-day battle against enemy forces.",
        "popularity": 1017.186,
        "posterPath": "https://image.tmdb.org/t/p/w185/hqnfqeCILvgKGWKOut5lVdxdeJh.jpg",
        "releaseDate": "2023-08-11",
        "title": "3 Days in Malay",
        "video": false,
        "voteAverage": 5.031,
        "voteCount": 16,
      },
      {
        "adult": false,
        "backdropPath": "https://image.tmdb.org/t/p/w185/wPNI1fd18z5TKBOK9Mv9Rfjb0j0.jpg",
        "genreIds": [
          53,
          18,
          28,
          80,
        ],
        "genres": undefined,
        "id": 729165,
        "originalLanguage": "en",
        "originalTitle": "Out of Exile",
        "overview": "Recently paroled thief Gabriel Russell tries to balance his life and mend a troubled family as an FBI agent hunts him down, along with his crew after a botched armored car robbery.",
        "popularity": 1006.516,
        "posterPath": "https://image.tmdb.org/t/p/w185/jgF5XaXnJmOgMxulhy2k1f9LNNc.jpg",
        "releaseDate": "2023-01-20",
        "title": "Out of Exile",
        "video": false,
        "voteAverage": 6.397,
        "voteCount": 29,
      },
      {
        "adult": false,
        "backdropPath": "https://image.tmdb.org/t/p/w185/2RVcJbWFmICRDsVxRI8F5xRmRsK.jpg",
        "genreIds": [
          27,
          878,
          53,
        ],
        "genres": undefined,
        "id": 762441,
        "originalLanguage": "en",
        "originalTitle": "A Quiet Place: Day One",
        "overview": "As New York City is invaded by alien creatures who hunt by sound, a woman named Sam fights to survive with her cat.",
        "popularity": 969.283,
        "posterPath": "https://image.tmdb.org/t/p/w185/yrpPYKijwdMHyTGIOd1iK1h0Xno.jpg",
        "releaseDate": "2024-06-26",
        "title": "A Quiet Place: Day One",
        "video": false,
        "voteAverage": 6.8,
        "voteCount": 547,
      },
      {
        "adult": false,
        "backdropPath": "https://image.tmdb.org/t/p/w185/59AJ2w9tKRSbBpnxKfB5UyIg6Jf.jpg",
        "genreIds": [
          27,
          53,
        ],
        "genres": undefined,
        "id": 1010600,
        "originalLanguage": "en",
        "originalTitle": "The Strangers: Chapter 1",
        "overview": "After their car breaks down in an eerie small town, a young couple are forced to spend the night in a remote cabin. Panic ensues as they are terrorized by three masked strangers who strike with no mercy and seemingly no motives.",
        "popularity": 910.449,
        "posterPath": "https://image.tmdb.org/t/p/w185/cdFZoQDxLRRWXLB3kxCnc674dBF.jpg",
        "releaseDate": "2024-05-15",
        "title": "The Strangers: Chapter 1",
        "video": false,
        "voteAverage": 5.976,
        "voteCount": 231,
      },
      {
        "adult": false,
        "backdropPath": "https://image.tmdb.org/t/p/w185/xk2y81O2FljJwuyFP1HoJnCO8ed.jpg",
        "genreIds": [
          878,
          28,
          12,
        ],
        "genres": undefined,
        "id": 823464,
        "originalLanguage": "en",
        "originalTitle": "Godzilla x Kong: The New Empire",
        "overview": "Following their explosive showdown, Godzilla and Kong must reunite against a colossal undiscovered threat hidden within our world, challenging their very existence – and our own.",
        "popularity": 886.302,
        "posterPath": "https://image.tmdb.org/t/p/w185/z1p34vh7dEOnLDmyCrlUVLuoDzd.jpg",
        "releaseDate": "2024-03-27",
        "title": "Godzilla x Kong: The New Empire",
        "video": false,
        "voteAverage": 7.188,
        "voteCount": 3304,
      },
      {
        "adult": false,
        "backdropPath": "https://image.tmdb.org/t/p/w185/z121dSTR7PY9KxKuvwiIFSYW8cf.jpg",
        "genreIds": [
          10752,
          28,
          18,
        ],
        "genres": undefined,
        "id": 929590,
        "originalLanguage": "en",
        "originalTitle": "Civil War",
        "overview": "In the near future, a group of war journalists attempt to survive while reporting the truth as the United States stands on the brink of civil war.",
        "popularity": 864.282,
        "posterPath": "https://image.tmdb.org/t/p/w185/sh7Rg8Er3tFcN9BpKIPOMvALgZd.jpg",
        "releaseDate": "2024-04-10",
        "title": "Civil War",
        "video": false,
        "voteAverage": 6.978,
        "voteCount": 2024,
      },
      {
        "adult": false,
        "backdropPath": "https://image.tmdb.org/t/p/w185/8fNBsXpgMvqk75SPsZzTmIqSRxY.jpg",
        "genreIds": [
          27,
          53,
        ],
        "genres": undefined,
        "id": 646683,
        "originalLanguage": "en",
        "originalTitle": "The Exorcism",
        "overview": "A troubled actor begins to unravel while shooting a supernatural horror film, leading his estranged daughter to wonder if he's slipping back into his past addictions or if there's something more sinister at play.",
        "popularity": 840.217,
        "posterPath": "https://image.tmdb.org/t/p/w185/ar2h87jlTfMlrDZefR3VFz1SfgH.jpg",
        "releaseDate": "2024-05-30",
        "title": "The Exorcism",
        "video": false,
        "voteAverage": 4.506,
        "voteCount": 88,
      },
      {
        "adult": false,
        "backdropPath": "https://image.tmdb.org/t/p/w185/xKJTWGvheOMMlTHgrjN18KaD9Ra.jpg",
        "genreIds": [
          27,
          53,
        ],
        "genres": undefined,
        "id": 1174618,
        "originalLanguage": "en",
        "originalTitle": "Alice in Terrorland",
        "overview": "A recently bereaved teenage girl goes to live with her aunt in a secluded woodland house, unaware that sinister forces lurk within.",
        "popularity": 825.12,
        "posterPath": "https://image.tmdb.org/t/p/w185/5XJGvr8g9jkmN6KUIOQOj2iE6K4.jpg",
        "releaseDate": "2023-12-26",
        "title": "Alice in Terrorland",
        "video": false,
        "voteAverage": 6.1,
        "voteCount": 111,
      },
    ],
    "meta": {
      "currentPage": 1,
      "firstPage": 1,
      "hasNextPage": true,
      "hasPreviousPage": false,
      "lastPage": 45331,
      "total": 906603,
    },
  },
}
`;
